### PR TITLE
[Backport] Changes required for snapshot missing in 1.1 

### PR DESCRIPTION
**Pull request by peternied**
### Description
Backporting all changes from main onto branch 1.1
* https://github.com/opensearch-project/common-utils/pull/58
* https://github.com/opensearch-project/common-utils/pull/57
 
### Issues
* https://github.com/opensearch-project/opensearch-build/issues/410

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,16 +34,16 @@ jobs:
...
       # common-utils
       - name: Build and Test
         run: |
-          ./gradlew build -Dopensearch.version=1.1.0
+          ./gradlew build -Dopensearch.version=1.1.0-SNAPSHOT
 
       - name: Publish to Maven Local
         run: |
-          ./gradlew publishToMavenLocal -Dopensearch.version=1.1.0
+          ./gradlew publishToMavenLocal -Dopensearch.version=1.1.0-SNAPSHOT
```
Comment (peterzhuamazon) [inline .github/workflows/ci.yml:46]: Why do we add all the snapshots back?

File: build.gradle [hunk 2]
```diff
@@ -12,7 +12,7 @@
 buildscript {
     ext {
         opensearch_group = "org.opensearch"
-        opensearch_version = System.getProperty("opensearch.version", "1.1.0")
+        opensearch_version = System.getProperty("opensearch.version", "1.1.0-SNAPSHOT")
```
Comment (peterzhuamazon) [inline build.gradle:15]: Same snapshots

File: build.gradle [hunk 3]
```diff
@@ -44,7 +44,17 @@ repositories {
...
+    isSnapshot = "true" == System.getProperty("build.snapshot", "true")
+}
+
+allprojects {
+    group 'org.opensearch.commons'
+    version = opensearch_version - '-SNAPSHOT' + '.0'
+    if (isSnapshot) {
+        version += "-SNAPSHOT"
+    }
+}
```
Comment (peterzhuamazon) [inline build.gradle:57]: Same Why do we add all the snapshots back?

**Comment by peternied** [comment]
@peterzhuamazon I did not make any code changes, just merged the 2 snapshot related PRs from main into this branch, if we need to have a branch specific change I'll need someone from common-util to take the lead

**Comment by peternied** [comment]
@VachaShah Can you take a look?

**Comment by dblock** [comment]
We should merge **everything** in main -> 1.x.

**Comment by peternied** [comment]
I am going to leave this up to the branch owners to migrate

**Pull request closed without merging**